x/poolmanager/taker_fee.go

```diff
@@ -77,6 +77,28 @@ func (k Keeper) GetTradingPairTakerFee(ctx sdk.Context, denom0, denom1 string) (
 	return takerFee.Dec, nil
 }
 
+// GetAllTradingPairTakerFees returns all the custom taker fees for trading pairs.
+func (k Keeper) GetAllTradingPairTakerFees(ctx sdk.Context) ([]types.DenomPairTakerFee, error) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStoreReversePrefixIterator(store, types.DenomTradePairPrefix)
```

Comment:
What is the reason for this being a reverse iterator?

Reply:
It could really be either because order does not matter, but if you don't do this then the KVStore in the state exported testnet ends up being flipped order wise.

i.e) If we stored A,B,C,D, and we do normal iterator, we end up with D,C,B,A in the genesis struct, which then when we iterate over it to store each we end up storing D,C,B,A.  

This was shown in the TestExportGenesis test. I can change this to normal iterator since order doesn't matter, it just makes the test cleaner.